sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/LocalDocumentStorage.java

```diff
@@ -206,7 +211,13 @@ boolean deleteOffline(String table, DocumentWrapper<Void> documentWrapper, Write
      * @return true if storage update succeeded, false otherwise.
      */
     boolean deleteOffline(String table, String partition, String documentId, WriteOptions writeOptions) {
-        DocumentWrapper<Void> writeDocument = new DocumentWrapper<>(null, partition, documentId);
+        DocumentWrapper<Void> cachedDocument = read(table, partition, documentId, Void.class, null);
+        DocumentWrapper<Void> writeDocument;
+        if (cachedDocument.getError() != null && !cachedDocument.getError().getMessage().equals(FAILED_TO_DESERIALIZE_DOCUMENT)) {
```

Comment:
We have to read from the cache to get the current etag so that it can be preserved when we update the pending operation in the cache. We don't get a document class type on delete though, which is required for read. Passing `Void.class` on read works, but the document fails to deserialize (which doesn't matter because we're only interested in the etag in the metadata, which isn't impacted by the deserialization of the document content).

Reply:
That looks like an anti-pattern. Why do we try to deserialize something that we'll know will fail and then check for this error while all of this could be avoided altogether?

Reply:
I was basing it off the pattern in `createOrUpdateOffline` where there are different behaviors depending on if the read failed with a specific error and then whether it failed/succeeded generally. I can refactor to add a method that only reads the metadata

Reply:
Why not simply always call line 219? cachedDocument.getETag() will return null anyway on "error but not quite" so the flow would be the same, we probably don't need a if here, am I missing something?

Reply:
No you're right that makes sense, fixed it to remove the if.